Machine-Teaching-Examples/moab_pomdp_esn.ink

```diff
@@ -0,0 +1,147 @@
+    ball_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+
+    # Ball X,Y velocity
+    ball_vel_x: number<-MaxVelocity .. MaxVelocity>,
+    ball_vel_y: number<-MaxVelocity .. MaxVelocity>,
+}
+
+# Imagine there isn't a sensor for measuring velocity and so we only provide
+# the brain with position information
+# This turns this problem into a Partially Observable Markov Decision Process (POMDP)
+type ObservableState {
+    # Ball X,Y position
+    ball_x: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+    ball_y: number<-MaxDistancePerStep - RadiusOfPlate .. RadiusOfPlate + MaxDistancePerStep>,
+}
+
+# Action provided as output by policy and sent as
+# input to the simulator
+type SimAction {
+    # Range -1 to 1 is a scaled value that represents
+    # the full plate rotation range supported by the hardware.
+    input_pitch: number<-1 .. 1>, # rotate about x-axis
+    input_roll: number<-1 .. 1>, # rotate about y-axis
+}
+
+# Per-episode configuration that can be sent to the simulator.
+# All iterations within an episode will use the same configuration.
+type SimConfig {
+    # Model initial ball conditions
+    initial_x: number<-RadiusOfPlate .. RadiusOfPlate>, # in (m)
+    initial_y: number<-RadiusOfPlate .. RadiusOfPlate>,
+
+    # Model initial ball velocity conditions
+    initial_vel_x: number<-MaxInitialVelocity .. MaxInitialVelocity>, # in (m/s)
+    initial_vel_y: number<-MaxInitialVelocity .. MaxInitialVelocity>,
+
+    # Range -1 to 1 is a scaled value that represents
+    # the full plate rotation range supported by the hardware.
+    initial_pitch: number<-1 .. 1>,
+    initial_roll: number<-1 .. 1>,
+}
+
+# Define a concept graph with a single concept
+graph (input: ObservableState) {
+    concept MoveToCenter(input): SimAction {
+        curriculum {
+            # The source of training for this concept is a simulator that
+            #  - can be configured for each episode using fields defined in SimConfig,
+            #  - accepts per-iteration actions defined in SimAction, and
+            #  - outputs states with the fields defined in SimState.
+            source simulator MoabSim(Action: SimAction, Config: SimConfig): SimState {
+                # Automatically launch the simulator with this
+                # registered package name.
+                package "Moab"
+            }
+
+            training {
+                # Limit episodes to 250 iterations instead of the default 1000.
+                EpisodeIterationLimit: 250
+            }
+
+            algorithm {
+                # Use supported values:
+                # default - AI engine will choose automatically
+                # none - no memory, learned actions depend on current state
+                # state - memory of past states
+                # state and action - memory of past states and actions.
+                MemoryMode: "state"
```

Comment:
this is fine, but note that once the memory feature is fully turned on, memory will be on by default, so users won't need to do anything to use it. 

Which makes me wonder what users will get out of this sample — the fact that it can learn without velocity is interesting....

Reply:
FYI our default (which is also a recommended mode to begin with) is "state and action", not just state.